lib/wallaby/phantom/driver.ex

```diff
@@ -333,6 +333,135 @@ defmodule Wallaby.Phantom.Driver do
+  """
+  def accept_alert(%Session{}=session, fun) do
+    script = """
+    var page = this;
+    page.__onAlertDefault = page.onAlert;
+    page.onAlert = function(msg) {
+      page.__alertMessage = msg;
+      page.onAlert = page.__onAlertDefault;
+    }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script)
+    fun.(session)
+    script = """
+    var page = this;
+    page.onAlert = page.__onAlertDefault;
+    return ["ok", page.__alertMessage];
+    """
+    {:ok, ["ok", message]} = execute_phantom_script(session, script)
+    message
+  end
+
+  @doc """
+  Accept one confirm triggered within `fun` and return the confirm message.
+  """
+  def accept_confirm(%Session{}=session, fun) do
+    handle_confirm(session, fun, true)
+  end
+
+  @doc """
+  Dismiss one confirm triggered within `fun` and return the confirm message.
+  """
+  def dismiss_confirm(%Session{}=session, fun) do
+    handle_confirm(session, fun, false)
+  end
+
+  defp handle_confirm(%Session{}=session, fun, return_value) do
+    script = """
+    var page = this, returnVal = arguments[0];
+    page.__onConfirmDefault = page.onConfirm;
+    page.onConfirm = function(msg) {
+      page.__confirmMessage = msg;
+      page.onConfirm = page.__onConfirmDefault;
+      return returnVal;
+    }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script, [return_value])
+    fun.(session)
+    script = """
+    var page = this;
+    page.onConfirm = page.__onConfirmDefault;
+    return ["ok", page.__confirmMessage];
+    """
+    {:ok, ["ok", message]} = execute_phantom_script(session, script)
+    message
+  end
+
+  @doc """
+  Accept one prompt triggered within `fun` with the specified `input_value`
+  and return the confirm message.
+  """
+  def accept_prompt(%Session{}=session, input_value, fun) do
+    handle_prompt(session, fun, input_value, true)
+  end
+
+  @doc """
+  Dismiss one prompt triggered within `fun` and return the confirm message.
+  """
+  def dismiss_prompt(%Session{}=session, fun) do
+    handle_prompt(session, fun, nil, false)
+  end
+
+  defp handle_prompt(%Session{}=session, fun, return_value, use_default) do
+    script = """
+    var page = this, returnVal = arguments[0], useDefault = arguments[1];
+    page.__onPromptDefault = page.onPrompt;
+    page.onPrompt = function(msg, defaultVal) {
+      page.__promptMessage = msg;
+      page.onPrompt = page.__onPromptDefault;
+      if (useDefault) {
+        return returnVal || defaultVal;
+      } else {
+        return returnVal;
+      }
+    }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script, [return_value, use_default])
+    fun.(session)
+    script = """
+    var page = this;
+    page.onPrompt = page.__onPromptDefault;
+    return ["ok", page.__promptMessage];
+    """
+    {:ok, ["ok", message]} = execute_phantom_script(session, script)
```

Comment:
huh that seems unexpected. Not a huge fans of code comments most of the time, but this would be a great place for a WHY comment (like the one you wrote up just now) so we don't stumble upon it along the road :)